js.go

```diff
@@ -399,6 +399,33 @@ type NextRequest struct {
 	NoWait  bool       `json:"no_wait,omitempty"`
 }
 
+// jsSub includes JetStream subscription info.
+type jsSub struct {
+	js       *js
+	consumer string
+	stream   string
+	deliver  string
+	pull     int
+	durable  bool
+	attached bool
+}
+
+func (jsi *jsSub) unsubscribe(drainMode bool) error {
+	if drainMode && (jsi.durable || jsi.attached) {
+		// Skip deleting consumer for durables/attached
+		// consumers when using drain mode.
+		return nil
+	}
+
+	// Skip if in direct mode as well.
+	js := jsi.js
+	if js.direct {
+		return nil
+	}
+
+	return js.DeleteConsumer(jsi.stream, jsi.consumer)
```

Comment:
This call fails with timeout during unsubscribe and `nc.Drain()` sometimes because in case the request mux handler got drained first then it will stop being able to receive responses. Trying to fix in another PR.